packages/types/src/client.ts

```diff
@@ -217,11 +223,13 @@ export interface Client<O extends ClientOptions = ClientOptions> {
    */
   emit?(hook: 'beforeEnvelope', envelope: Envelope): void;
 
-  /*
-   * Fire a hook event before sending an event. Expects to be given an Event & EventHint as the
-   * second/third argument.
+  /**
+   * Fire a hook event before sending an event.
+   * `beforeSendEvent` is called right before an event is sent and should not be used to mutate the event.
+   * `preprocessEvent` is called before all global event processors.
+   * Expects to be given an Event & EventHint as the second/third argument.
    */
-  emit?(hook: 'beforeSendEvent', event: Event, hint?: EventHint): void;
```

Comment:
Can we make this two separate hooks? If necessary by ignoring the lint rules. Upside: Separate JSDoc.

Reply:
I split them up and ignored this eslint rule there!